src/pow.cpp

```diff
@@ -38,16 +38,39 @@ void ResetChallenge(CBlockHeader& block, const CBlockIndex& indexLast, const Con
     block.proof.challenge = indexLast.proof.challenge;
 }
 
-bool CheckBitcoinProof(uint256 hash, unsigned int nBits)
+static const unsigned int BLOCK_VERIFY_FLAGS = MANDATORY_SCRIPT_VERIFY_FLAGS |
+                                               SCRIPT_VERIFY_DERSIG |
+                                               SCRIPT_VERIFY_STRICTENC |
+                                               SCRIPT_VERIFY_MINIMALDATA |
+                                               SCRIPT_VERIFY_NULLDUMMY |
+                                               SCRIPT_VERIFY_DISCOURAGE_UPGRADABLE_NOPS |
+                                               SCRIPT_VERIFY_CLEANSTACK |
+                                               SCRIPT_VERIFY_MINIMALIF |
+                                               SCRIPT_VERIFY_NULLFAIL |
+                                               SCRIPT_VERIFY_CHECKLOCKTIMEVERIFY |
+                                               SCRIPT_VERIFY_CHECKSEQUENCEVERIFY |
+                                               SCRIPT_VERIFY_LOW_S |
+                                               SCRIPT_VERIFY_WITNESS |
+                                               SCRIPT_VERIFY_DISCOURAGE_UPGRADABLE_WITNESS_PROGRAM |
+                                               SCRIPT_VERIFY_WITNESS_PUBKEYTYPE;
+
+bool CheckBitcoinProof(const uint256& hash, unsigned int nBits, const Consensus::Params& params)
 {
+    if (g_solution_blocks) {
+        if (hash == params.hashGenesisBlock) return true;
```

Comment:
I guess we don't need or want this here.

Reply:
You would need to provide the signature for the genesis block otherwise, no?

Reply:
No, this is used to validate the parent chain. The sidechain doesn't validate the parent chain's genesis block (in fact, not even the parent chain should validate the genesis block, the genesis block is a consensus rule in itself and thus doesn't need to be validated, not when it's pow nor when it's blocksigned).